pkg/translator/prometheusremotewrite/metrics_to_prw.go

```diff
@@ -39,21 +38,37 @@ func FromMetrics(md pmetric.Metrics, settings Settings) (map[string]*prompb.Time
 	return out, errs
 }
 
-// prometheusConverter converts from OTel write format to Prometheus write format.
-type prometheusConverter struct {
+// PrometheusConverter converts from OTel write format to Prometheus write format.
+// Internally it keeps a buffer of labels to avoid expensive allocations, so it is
+// best to keep it around for the lifetime of the Go process. Due to this shared
+// state, PrometheusConverter is NOT thread-safe and is only intended to be used by
+// a single go-routine at a time.
+// Each FromMetrics call should be followed by a Reset when the metrics can be safely
```

Comment:
Should we emit a warning log or something if someone calls FromMetrics without Resetting?

Reply:
Moved the reset to always be called inside FromMetrics so this is no longer a user concern.

Reply:
Now that the user doesn't need to call reset, should we remove this part of the comment?
`// Each FromMetrics call should be followed by a Reset.....`